### PR TITLE
Refactor BetterTransformerManager requirement validation methods

### PR DESCRIPTION
**Pull request by baskrahmer**
# What does this PR do?

Refactoring of methods to make them less ugly ;)

### CODE REVIEW

File: optimum/bettertransformer/models/__init__.py [hunk 1]
```diff
@@ -118,6 +118,49 @@ class BetterTransformerManager:
         "glpn": "GLPN has a convolutional layer present in the FFN network, which is not supported in PyTorch's BetterTransformer.",
     }
 
+    REQUIRES_NESTED_TENSOR = {
```
Comment (ydshieh) [inline optimum/bettertransformer/models/__init__.py:121]: The naming and the logic below is actually reversed.

```python
def requires_nested_tensor
    return model_type not in BetterTransformerManager.REQUIRES_NESTED_TENSOR
```

Comment (baskrahmer) [reply optimum/bettertransformer/models/__init__.py:121]: I don't follow. Can you give a corrected example?

Comment (ydshieh) [reply optimum/bettertransformer/models/__init__.py:121]: The name is `REQUIRES_NESTED_TENSOR`. But for a model_type in `REQUIRES_NESTED_TENSOR`, the method return `False`, but the method name is `requires_nested_tensor`. The name of the method and what it actually does is completely different thing, if I don't make mistake.

Comment (ydshieh) [reply optimum/bettertransformer/models/__init__.py:121]: One quick way is to rename `REQUIRES_NESTED_TENSOR` to `NOT_REQUIRES_NESTED_TENSOR`.

Comment (baskrahmer) [reply optimum/bettertransformer/models/__init__.py]: Ah gotcha! Good point, thanks for the feedback I will update the PR.

**Review by fxmarty — APPROVED** [review]
Thanks!

**Pull request merged**